contracts/strategies/DirectAllocation.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import {CoreBaseStrategy} from "./CoreBaseStrategy.sol";
+
+/// @title DirectAllocationStrategy
+/// @dev The strategy only implements the allocate logic
+/// @notice A strategy that directly allocates funds to a recipient
+contract DirectAllocationStrategy is CoreBaseStrategy {
+    /// ===============================
+    /// ============ Errors ===========
+    /// ===============================
+
+    /// @notice Error when the function is not implemented
+    error NOT_IMPLEMENTED();
+
+    /// @notice Error when the input is invalid
+    error INVALID_INPUT();
+
+    /// ===============================
+    /// ============ Events ===========
+    /// ===============================
+
+    /// @notice Emitted when direct allocating to a recipient
+    /// @param recipient The recipient
+    /// @param amount The amount allocated
+    /// @param token The token allocated
+    /// @param sender The sender
+    event DirectAllocated(address indexed recipient, uint256 amount, address token, address sender);
+
+    /// ===============================
+    /// ========= Constructor =========
+    /// ===============================
+
+    constructor(address _allo) CoreBaseStrategy(_allo) {}
+
+    /// ===============================
+    /// ========= Initialize ==========
+    /// ===============================
+
+    /// @notice Initialize the strategy
+    /// @param _poolId The pool id
+    /// @param _data The data to initialize the strategy
+    function initialize(uint256 _poolId, bytes memory _data) external virtual override {
+        __BaseStrategy_init(_poolId);
+        emit Initialized(_poolId, _data);
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    /// @notice Allocate funds to a recipient
+    /// @param _recipients The recipients
+    /// @param _amounts The amounts to allocate
+    /// @param _data The data to decode
+    /// @param _sender The sender
+    function _allocate(address[] memory _recipients, uint256[] memory _amounts, bytes memory _data, address _sender)
+        internal
+        virtual
+        override
+    {
+        /// Decode the data (data: tokens)
+        address[] memory _tokens = abi.decode(_data, (address[]));
+
+        uint256 _recipientsLength = _recipients.length;
+        /// Check if inputs match the decoded data
+        if (_recipientsLength != _amounts.length || _recipientsLength != _tokens.length) {
+            revert INVALID_INPUT();
+        }
+
+        for (uint256 _i = 0; _i < _recipientsLength;) {
+            /// Direct allocate the funds
+            _transferAmountFrom(_tokens[_i], TransferData({from: _sender, to: _recipients[_i], amount: _amounts[_i]}));
```

Comment:
I don't know if this strategy is expected to hold native tokens in any use case, but beware that `_transferAmountFrom()` won't check if msg.value is equal to the total amount.